index.js

```diff
@@ -1,255 +1,53 @@
-// External Libraries
-const agent = require('agentkeepalive');
-const async = require('async');
-const aws = require('aws-sdk');
-const request = require('request');
-const zlib = require('zlib');
+'use strict'
 
-// Constants
-const BATCH_INTERVAL_MS = parseInt(process.env.LOGDNA_BATCH_INTERVAL) || 50;
-const BATCH_LIMIT = parseInt(process.env.LOGDNA_BATCH_LIMIT) || 25;
-const FREE_SOCKET_TIMEOUT_MS = parseInt(process.env.LOGDNA_FREE_SOCKET_TIMEOUT) || 300000;
-const LOGDNA_URL = process.env.LOGDNA_URL || 'https://logs.logdna.com/logs/ingest';
-const MAX_REQUEST_RETRIES = parseInt(process.env.LOGDNA_MAX_REQUEST_RETRIES) || 5;
-const MAX_REQUEST_TIMEOUT_MS = parseInt(process.env.LOGDNA_MAX_REQUEST_TIMEOUT) || 30000;
-const REQUEST_RETRY_INTERVAL_MS = parseInt(process.env.LOGDNA_REQUEST_RETRY_INTERVAL) || 100;
-const INTERNAL_SERVER_ERROR = 500;
-const DEFAULT_HTTP_ERRORS = [
-    'ECONNRESET'
-    , 'EHOSTUNREACH'
-    , 'ETIMEDOUT'
-    , 'ESOCKETTIMEDOUT'
-    , 'ECONNREFUSED'
-    , 'ENOTFOUND'
-];
+const async = require('async')
 
-// RegExp:
-const DOT_REGEXP = /\./g;
+const config = require('./lib/config.js')
+const eventHandler = require('./lib/event-handler.js')
+const logger = require('./lib/logger.js')
+const transformer = require('./lib/transformer.js')
 
-// Initializations
-const s3 = new aws.S3({
-    apiVersion: '2006-03-01'
-});
+const BATCH_INTERVAL = parseInt(process.env.LOGDNA_BATCH_INTERVAL) || 50
 
-// Get Configuration from Environment Variables
-const getConfig = (event) => {
-    const pkg = require('./package.json');
-    let config = {
-        eventlog: false
-        , filelog: true
-        , hostname: event.Records[0].s3.bucket.name.replace(DOT_REGEXP, '_')
-        , UserAgent: `${pkg.name}/${pkg.version}`
-    };
+module.exports = {
+  handler
+}
 
-    if (process.env.LOGDNA_KEY) config.key = process.env.LOGDNA_KEY;
-    if (process.env.LOGDNA_HOSTNAME) config.hostname = process.env.LOGDNA_HOSTNAME;
-    if (process.env.LOGDNA_TAGS) {
-        config.tags = process.env.LOGDNA_TAGS.split(',').map(tag => tag.trim()).join(',');
-    }
 
-    if (process.env.LOGDNA_EVENTLOG) {
-        config.eventlog = process.env.LOGDNA_EVENTLOG.toLowerCase();
-        config.eventlog = config.eventlog.indexOf('yes') > -1 || config.eventlog.indexOf('true') > -1;
-    }
+function handler(event, context, callback) {
```

Comment:
these days, the `handler` can be an [async function](https://docs.aws.amazon.com/lambda/latest/dg/nodejs-handler.html), which would then allow you to cleanly use async/await everywhere.

Reply:
Will alter this in the next commit where I will replace `logger.flush` with `logger-node`